cms/server/admin/handlers/archive.py

```
@@ -0,0 +1,570 @@
+            )
+
+            # Archive attendance data for each student
+            self._archive_attendance_data(training_day, contest, class_ips)
+
+            # Archive ranking data for each student
+            self._archive_ranking_data(training_day, contest)
+
+            # Delete the contest (this will cascade delete participations)
+            self.sql_session.delete(contest)
+
+        except Exception as error:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)